plotly/tools.py

```diff
@@ -1443,30 +1443,320 @@ class FigureFactory(object):
+            else:
+                height = 1231
+        if (width is None):
+            if dim <= 4:
+                width = 985
+            else:
+                width = 1231
+            
+        if diag == "Histogram":
+            fig['layout'].update(
+                height = height, width = width,
+                title = title,
+                showlegend = True,
+                barmode = "stack")
+            return fig
+
+        elif diag == "box":
+            fig['layout'].update(
+                height = height, width = width,
+                title = title,
+                showlegend = True)
+            return fig
+
+        else:
+            fig['layout'].update(
+                height = height, width = width,
+                title = title,
+                showlegend = True)
+            return fig
+
+
+    @staticmethod
+    def _validate_index(index_vals):
+        import types
+        NumberTypes = (types.IntType, types.LongType, types.FloatType)
+        if isinstance(index_vals[0], NumberTypes):
+            if not all(isinstance(item, NumberTypes) for item in index_vals):
+                raise exceptions.PlotlyError("Error in indexing column."
+                                             "Make sure all entries of each "
+                                             "column are all numbers or "
+                                             "all strings.")
+
+        elif isinstance(index_vals[0], basestring):
+            if not all(isinstance(item, basestring) for item in index_vals):
+                raise exceptions.PlotlyError("Error in indexing column."
+                                             "Make sure all entries of each "
+                                             "column are all numbers or "
+                                             "all strings.")
+
+    @staticmethod
+    def _validate_matrix(array):
+        import types
+        NumberTypes = (types.IntType, types.LongType, types.FloatType)
+        for vector in array:
+            if isinstance(vector[0], NumberTypes):
+                if not all(isinstance(item, NumberTypes) for item in vector):
+                    raise exceptions.PlotlyError("Error in dataframe. "
+                                                 "Make sure all entries of "
+                                                 "each column are either "
+                                                 "numbers or strings.")
+            elif isinstance(vector[0], basestring):
+                if not all(isinstance(item, basestring) for item in vector):
+                    raise exceptions.PlotlyError("Error in dataframe. "
+                                                 "Make sure all entries of "
+                                                 "each column are either "
+                                                 "numbers or strings.")
+
+    @staticmethod
+    def _validate_scatterplotmatrix(df, jitter, index, diag, **scatter_kws):
```

Comment:
Typically it makes code more readable not to change the names of `*args` and `**kwargs` in call signatures.